Expose program arguements to the profile

boraozarslan opened the pull request:
Exposes the program arguements starting from the profile name by appending th arguements
to the environment variables list. For example in this usage:
	./capsicumizer ls.profile ./directory

$0 = ls.profile
$1 = ./directory

valpackett commented on README.md:
````diff
@@ -61,7 +61,7 @@ ld_preload = [
 ```
 
-Environment variables are exposed as UCL variables.
+Environment variables and program arguements ($0 etc.) are exposed as UCL variables.
````
typo. should be `arguments`

boraozarslan replied:
Will fix

myfreeweb commented:
I'm not sure how this could be useful, do you have a use case in mind?

This doesn't look like it has anything to do with #1, that one was fixed by https://github.com/musec/libpreopen/pull/10 and was not related to arguments at all

boraozarslan committed:
Expose program arguments to the profile

Exposes the program arguments starting from the profile name by appending th arguments
to the environment variables list. For example in this usage:
	./capsicumizer ls.profile ./directory

$0 = ls.profile
$1 = ./directory

boraozarslan commented:
I meant to tag #2, sorry. I have cases where a specific argument is always needed to be in the list of directories to be opened. I know it can be done by inserting the directory to the profile each time but I find it more convenient to be able to specify: "I want argument 2 to always be preopened". For example, if I were to run 'ls' I would want $1 to always be opened.

myfreeweb commented:
oh. #2 is kind of the opposite :)

Yeah, so the problem with that is, we don't know if $1 would be a directory path or the flags… We also don't know how many path arguments there are.

I guess this patch can be useful (just remove any 'resolve #x' from the commit message), but there should be some better way of handling this.

boraozarslan commented:
I'm not sure what other way of doing that is possible besides evaluating the args the same way as the underlying program. Perhaps one can write the signature of the program and capsicumizer can do something like pattern matching on what the actual args are.

myfreeweb commented:
hm, also, it's all going to be shifted by one for `./capsicumizer ls.profile` vs `./ls.profile` (using `#!`)
